src/System.Net.Requests/ref/System.Net.Requests.cs

```diff
@@ -48,11 +67,186 @@ public partial interface IWebRequestCreate
+    {
+        internal OpenWriteCompletedEventArgs(System.IO.Stream result, System.Exception exception, bool cancelled, object userToken) : base(exception, cancelled, userToken) { }
+        public System.IO.Stream Result { get { throw null; } }
+    }
     public partial class ProtocolViolationException : System.InvalidOperationException
     {
         public ProtocolViolationException() { }
         public ProtocolViolationException(string message) { }
     }
+    public class UploadDataCompletedEventArgs : System.ComponentModel.AsyncCompletedEventArgs
+    {
+        internal UploadDataCompletedEventArgs(byte[] result, System.Exception exception, bool cancelled, object userToken) : base(exception, cancelled, userToken) { }
+        public byte[] Result { get { throw null; } }
+    }
+    public delegate void UploadDataCompletedEventHandler(object sender, System.Net.UploadDataCompletedEventArgs e);
+    public class UploadFileCompletedEventArgs : System.ComponentModel.AsyncCompletedEventArgs
+    {
+        internal UploadFileCompletedEventArgs(byte[] result, System.Exception exception, bool cancelled, object userToken) : base(exception, cancelled, userToken) { }
+        public byte[] Result { get { throw null; } }
+    }
+    public delegate void UploadFileCompletedEventHandler(object sender, System.Net.UploadFileCompletedEventArgs e);
+    public class UploadProgressChangedEventArgs : System.ComponentModel.ProgressChangedEventArgs
+    {
+        internal UploadProgressChangedEventArgs(int progressPercentage, object userToken, long bytesSent, long totalBytesToSend, long bytesReceived, long totalBytesToReceive) : base(progressPercentage, userToken) { }
+        public long BytesReceived { get { throw null; } }
+        public long TotalBytesToReceive { get { throw null; } }
+        public long BytesSent { get { throw null; } }
+        public long TotalBytesToSend { get { throw null; } }
+    }
+    public delegate void UploadProgressChangedEventHandler(object sender, System.Net.UploadProgressChangedEventArgs e);
+    public class UploadStringCompletedEventArgs : System.ComponentModel.AsyncCompletedEventArgs
+    {
+        internal UploadStringCompletedEventArgs(string result, System.Exception exception, bool cancelled, object userToken) : base(exception, cancelled, userToken) { }
+        public string Result { get { throw null; } }
+    }
+    public delegate void UploadStringCompletedEventHandler(object sender, System.Net.UploadStringCompletedEventArgs e);
+    public class UploadValuesCompletedEventArgs : System.ComponentModel.AsyncCompletedEventArgs
+    {
+        internal UploadValuesCompletedEventArgs(byte[] result, System.Exception exception, bool cancelled, object userToken) : base(exception, cancelled, userToken) { }
+        public byte[] Result { get { throw null; } }
+    }
+    public delegate void UploadValuesCompletedEventHandler(object sender, System.Net.UploadValuesCompletedEventArgs e);
+    public class WebClient
```

Comment:
I don't understand why this isn't in its own contract now.  I was thinking this should be in a separate contract, `System.Net.WebClient`.  Yes, it uses *WebRequest APIs but it should be using them only implementation purposes with the current implementation. But that could change.